query/parser.go

```diff
@@ -340,8 +384,12 @@ func (p *Parser) insertPropertyKeyValue() {
 	case "from":
 		fallthrough
 	case "to":
-		unix, err := parseDate(value)
-		if err != nil {
+		var unix int64
+		var err error
+		if unix, err = parseDate(value); err == nil {
+			// Valid, so do nothing.
+		} else if unix, err = parseRelativeTime(value, time.Now().UTC()); err != nil {
```

Comment:
To address (1) you could compute `time.Now()` just once, prior to construction of the `EvaluationContext`, and then pass it as a parameter here.

For (2) you could do something similar, only allow an override for `now` specified by the client/query (who calculate it the first time, then save the query alongside the "now" timestamp to reproduce the original query).